{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1385933e-474a-41d3-bc80-d0300fc9da74",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b4fc6991-3304-4bd3-870c-13bd34fa1cee",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_path = \"/Users/tennisnyjmac/Desktop/BI/bioinformatics_institute/BI_practicum/AntibioticResistance_HW1/VarScan_results_annotated.vcf\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "c3b8744f-a075-46a3-a843-1e208cc4152e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G|missense_variant|MODERATE|ftsI|b0084|transcript|b0084|protein_coding|1/1|c.1631C>G|p.Ala544Gly|1631/1767|1631/1767|544/588||\n",
      "A|missense_variant|MODERATE|acrB|b0462|transcript|b0462|protein_coding|1/1|c.1706A>T|p.Gln569Leu|1706/3150|1706/3150|569/1049||\n",
      "G|upstream_gene_variant|MODIFIER|glnH|b0811|transcript|b0811|protein_coding||c.-4758T>C|||||4758|\n",
      "A|missense_variant|MODERATE|mntP|b1821|transcript|b1821|protein_coding|1/1|c.74G>A|p.Gly25Asp|74/567|74/567|25/188||\n",
      "C|missense_variant|MODERATE|envZ|b3404|transcript|b3404|protein_coding|1/1|c.722T>G|p.Val241Gly|722/1353|722/1353|241/450||\n",
      "T|synonymous_variant|LOW|rsgA|b4161|transcript|b4161|protein_coding|1/1|c.756C>A|p.Ala252Ala|756/1053|756/1053|252/350||WARNING_TRANSCRIPT_NO_START_CODON\n",
      "    Chromosome Position Reference Allele Alternative Allele    Impact  \\\n",
      "0  NC_000913.3    93043                C                  G  MODERATE   \n",
      "1  NC_000913.3   482698                T                  A  MODERATE   \n",
      "2  NC_000913.3   852762                A                  G  MODIFIER   \n",
      "3  NC_000913.3  1905761                G                  A  MODERATE   \n",
      "4  NC_000913.3  3535147                A                  C  MODERATE   \n",
      "5  NC_000913.3  4390754                G                  T       LOW   \n",
      "\n",
      "  Gene Name Gene ID Transcript ID           Variant Type Amino Acid Change  \\\n",
      "0      ftsI   b0084         b0084       missense_variant       p.Ala544Gly   \n",
      "1      acrB   b0462         b0462       missense_variant       p.Gln569Leu   \n",
      "2      glnH   b0811         b0811  upstream_gene_variant                     \n",
      "3      mntP   b1821         b1821       missense_variant        p.Gly25Asp   \n",
      "4      envZ   b3404         b3404       missense_variant       p.Val241Gly   \n",
      "5      rsgA   b4161         b4161     synonymous_variant       p.Ala252Ala   \n",
      "\n",
      "  cDNA Position Protein Position Genotype Depth  \n",
      "0     c.1631C>G        1631/1767      1/1    17  \n",
      "1     c.1706A>T        1706/3150      1/1    16  \n",
      "2    c.-4758T>C                       1/1    14  \n",
      "3       c.74G>A           74/567      1/1    13  \n",
      "4      c.722T>G         722/1353      1/1    17  \n",
      "5      c.756C>A         756/1053      1/1    15  \n"
     ]
    }
   ],
   "source": [
    "parsed_data = []\n",
    "\n",
    "with open(file_path, \"r\") as file:\n",
    "    for line in file:\n",
    "        if not line.startswith(\"#\"):\n",
    "            line = line.split('\\t')\n",
    "            chromosome = line[0]\n",
    "            position = line[1]\n",
    "            ref_allele = line[3]\n",
    "            alt_allele = line[4]\n",
    "            info_field = line[7]\n",
    "            format_field = line[8]\n",
    "            sample_field = line[9]\n",
    "\n",
    "            ann_field = [field for field in info_field.split(';') if field.startswith('ANN=')]\n",
    "            if ann_field:\n",
    "                annotation = ann_field[0].replace('ANN=', '').split(',')[0]\n",
    "                print(annotation)\n",
    "                ann_split = annotation.split('|')\n",
    "                parsed_data.append({\n",
    "                    'Chromosome': chromosome,\n",
    "                    'Position': position,\n",
    "                    'Reference Allele': ref_allele,\n",
    "                    'Alternative Allele': alt_allele,\n",
    "                    'Impact': ann_split[2],  # Example: MODERATE, LOW, etc.\n",
    "                    'Gene Name': ann_split[3],\n",
    "                    'Gene ID': ann_split[4],\n",
    "                    'Transcript ID': ann_split[6],\n",
    "                    'Variant Type': ann_split[1],  # Example: missense_variant, upstream_gene_variant, etc.\n",
    "                    'Amino Acid Change': ann_split[10],\n",
    "                    'cDNA Position': ann_split[9],\n",
    "                    'Protein Position': ann_split[11],\n",
    "                    'Genotype': sample_field.split(':')[0],  # Extract genotype from FORMAT\n",
    "                    'Depth': sample_field.split(':')[3]  # Extract depth (DP) from FORMAT\n",
    "                })\n",
    "            \n",
    "df = pd.DataFrame(parsed_data)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "564d35c0-f06c-427b-bf6f-d2e97a9a80ac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Reference Allele</th>\n",
       "      <th>Alternative Allele</th>\n",
       "      <th>Impact</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>Transcript ID</th>\n",
       "      <th>Variant Type</th>\n",
       "      <th>Amino Acid Change</th>\n",
       "      <th>cDNA Position</th>\n",
       "      <th>Protein Position</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Depth</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>93043</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>ftsI</td>\n",
       "      <td>b0084</td>\n",
       "      <td>b0084</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Ala544Gly</td>\n",
       "      <td>c.1631C&gt;G</td>\n",
       "      <td>1631/1767</td>\n",
       "      <td>1/1</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>482698</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>acrB</td>\n",
       "      <td>b0462</td>\n",
       "      <td>b0462</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Gln569Leu</td>\n",
       "      <td>c.1706A&gt;T</td>\n",
       "      <td>1706/3150</td>\n",
       "      <td>1/1</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>852762</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>glnH</td>\n",
       "      <td>b0811</td>\n",
       "      <td>b0811</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td></td>\n",
       "      <td>c.-4758T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>1/1</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>1905761</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>mntP</td>\n",
       "      <td>b1821</td>\n",
       "      <td>b1821</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Gly25Asp</td>\n",
       "      <td>c.74G&gt;A</td>\n",
       "      <td>74/567</td>\n",
       "      <td>1/1</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>3535147</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>envZ</td>\n",
       "      <td>b3404</td>\n",
       "      <td>b3404</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Val241Gly</td>\n",
       "      <td>c.722T&gt;G</td>\n",
       "      <td>722/1353</td>\n",
       "      <td>1/1</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>4390754</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>LOW</td>\n",
       "      <td>rsgA</td>\n",
       "      <td>b4161</td>\n",
       "      <td>b4161</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>p.Ala252Ala</td>\n",
       "      <td>c.756C&gt;A</td>\n",
       "      <td>756/1053</td>\n",
       "      <td>1/1</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Chromosome Position Reference Allele Alternative Allele    Impact  \\\n",
       "0  NC_000913.3    93043                C                  G  MODERATE   \n",
       "1  NC_000913.3   482698                T                  A  MODERATE   \n",
       "2  NC_000913.3   852762                A                  G  MODIFIER   \n",
       "3  NC_000913.3  1905761                G                  A  MODERATE   \n",
       "4  NC_000913.3  3535147                A                  C  MODERATE   \n",
       "5  NC_000913.3  4390754                G                  T       LOW   \n",
       "\n",
       "  Gene Name Gene ID Transcript ID           Variant Type Amino Acid Change  \\\n",
       "0      ftsI   b0084         b0084       missense_variant       p.Ala544Gly   \n",
       "1      acrB   b0462         b0462       missense_variant       p.Gln569Leu   \n",
       "2      glnH   b0811         b0811  upstream_gene_variant                     \n",
       "3      mntP   b1821         b1821       missense_variant        p.Gly25Asp   \n",
       "4      envZ   b3404         b3404       missense_variant       p.Val241Gly   \n",
       "5      rsgA   b4161         b4161     synonymous_variant       p.Ala252Ala   \n",
       "\n",
       "  cDNA Position Protein Position Genotype Depth  \n",
       "0     c.1631C>G        1631/1767      1/1    17  \n",
       "1     c.1706A>T        1706/3150      1/1    16  \n",
       "2    c.-4758T>C                       1/1    14  \n",
       "3       c.74G>A           74/567      1/1    13  \n",
       "4      c.722T>G         722/1353      1/1    17  \n",
       "5      c.756C>A         756/1053      1/1    15  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.to_csv(\"SNV_annotation_snpEff.csv\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
